task_geo/data_sources/hdx_oxfd/hdx_oxfd_formatter.py

```diff
@@ -0,0 +1,96 @@
+import logging
+from datetime import datetime
+
+import pandas as pd
+
+from task_geo.common.country_codes import iso2_to_iso3, iso3_to_iso2
+
+logger = logging.getLogger(__name__)
+
+
+def hdx_oxfd_formatter(raw, start_date, end_date, country_iso2):
+    """Formats raw pandas.DataFrame
+    - Drops columns ending with _note
+    - Orders Columns
+    - Converts ISO3 to ISO2 country codes
+
+    Arguments:
+        raw (pandas.DataFrame): from hdx_acap_connector
+
+    Returns: pandas.DataFrame
+
+    """
+    data = raw.copy()
+    data.columns = [column.lower() for column in data.columns]
+    # Remove columns with notes
+    drop_columns = [column for column in data.columns if column.find("_note") > -1]
+    data = data.drop(drop_columns, axis=1)
+    # Filter by country(s)
+    iso3 = []
+    for c in country_iso2:
+        iso3.append(iso2_to_iso3(c))
+    data = filter_by_country(data, iso3)
+    # Filter by date
+    data = filter_with_date(data, start_date, end_date)
```

Comment:
This is a conceptual error:

There are data sources that accept start_date and end_date or country, but this is because the source they are connecting demands this as argument or because the main dataset (NOAA FTP, for example starts in 1850) has data for time intervals which are not relevant.

Beyond that I see that the dates are passed as strings which is a bad practice, if you need a date, then the arguments should be `datetime` or `date`.

Please delete this two lines.

Reply:
deleted